internal/build/docker.go

```diff
@@ -95,6 +99,42 @@ func (l *localDockerBuilder) buildBaseWithMounts(ctx context.Context, baseDocker
 	return getDigestFromOutput(output.String())
 }
 
+func (l *localDockerBuilder) buildImageWithSteps(ctx context.Context, baseDigest digest.Digest, tag string, steps []Cmd) (digest.Digest, error) {
+	imageWithSteps := baseDigest
+	for _, s := range steps {
+		resp, err := l.dcli.ContainerCreate(ctx, &container.Config{
```

Comment:
guhhh it seems like there should be a way to create a single container and run all the steps on it in sequence rather than creating a container per step? maybe if you `&&`'d them all together into a single bash command?